### ISSUE TITLE
processArguments not work on 

### ISSUE THREAD

**Issue by camillake** [appium/appium#5909]
# I want send additional variables, "-e AlwaysShowTutorial 1", to my App, but *additional quotes mark* are added to the command line, causing instruments doesn't accept additional variables.#

**Appium-Python-Client v0.17
Appium v1.4.11
 development platform: Mac OS 10.10.5
 target device platform: iOS 8.4.1 iPad2
 Xcode 7.0**

# Log 
info: Welcome to Appium v1.4.11 (REV 8cf8311f00e59a2b10fde1834fcf6d5ace6fbcd0)
info: Appium REST http interface listener started on 127.0.0.1:4723
info: [debug] Non-default server args: {"address":"127.0.0.1"}
info: Console LogLevel: debug
info: --> POST /wd/hub/session {"desiredCapabilities":{"launchTimeout":"40000","platformVersion":"8.4.1","deviceName":"iPad4","processArguments":"-e AlwaysShowTutorial 1","noReset":true,"appPackage":null,"newCommandTimeout":"60","autoLaunch":true,"app":null,"platformName":"iOS"}}
info: Client User-Agent string: Python-urllib/3.4

**info: [debug] The following desired capabilities were provided, but not recognized by appium. They will be passed on to any other services running on this server. : processArguments**

info: [debug] App is an iOS bundle, will attempt to run as pre-existing
info: [debug] Creating new appium session c9ee137e-fc3e-48e3-b102-65fa9f1f3b97
info: [debug] Removing any remaining instruments sockets
info: [debug] Cleaned up instruments socket /tmp/instruments_sock
info: [debug] Auto-detecting iOS udid...
info: [debug] Not auto-detecting udid, running on sim
info: [debug] Localizable.strings is not currently supported when using real devices.
info: [debug] Creating instruments
info: [debug] Preparing uiauto bootstrap
info: [debug] Dynamic bootstrap dir: /Users/Tester/Library/Application Support/appium/bootstrap
info: [debug] Dynamic env: {"nodePath":"/usr/local/Cellar/node/4.1.0/bin/node","commandProxyClientPath":"/usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js","instrumentsSock":"/tmp/instruments_sock","interKeyDelay":null,"justLoopInfinitely":false,"autoAcceptAlerts":false,"autoDismissAlerts":false,"sendKeyStrategy":"grouped"}
...o: [debug] Dynamic bootstrap code: // This file is automatically generated. Do not manually modify!
info: [debug] Dynamic bootstrap path: /Users/Tester/Library/Application Support/appium/bootstrap/bootstrap-92788c7fc5a72907.js
info: [debug] Reusing dynamic bootstrap: /Users/Tester/Library/Application Support/appium/bootstrap/bootstrap-92788c7fc5a72907.js
info: [debug] Attempting iOS device log capture via libimobiledevice idevicesyslog
info: [debug] App is installed.
info: [debug] fullReset not requested. No need to install.
info: [debug] Starting command proxy.
info: [debug] Instruments socket server started at /tmp/instruments_sock
info: [debug] Starting instruments
info: [debug] Instruments is at: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments
info: Launching instruments
info: [debug] Attempting to run app on real device with UDID b51ba2976173fb65c26e9cd72029a719b0afd014
info: [debug] Attempting to run app with process arguments: -e AlwaysShowTutorial 1
info: On xcode 7.0, instruments-without-delay does not work, skippinginstruments-without-delay

**info: [debug] Spawning instruments with command: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments -t /Applications/Xcode.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate -D /tmp/appium-instruments/instrumentscli0.trace -w b51ba2976173fb65c26e9cd72029a719b0afd014 com.test.test1"-e AlwaysShowTutorial 1" -e UIASCRIPT "/Users/Tester/Library/Application Support/appium/bootstrap/bootstrap-92788c7fc5a72907.js" -e UIARESULTSPATH /tmp/appium-instruments**

**Comment by Jonahss** [comment]
Looks like a bug.
Could you try on appium version 1.5 (still in beta but will be published soon). You can do `npm install appium@beta`). If the error is still there we will fix it.

**Comment by camillake** [comment]
Sorry to say that still the additional quotes mark added.

**Comment by camillake** [comment]
Hello,
Any updates?

**Comment by jlipps** [comment]
@camillake you are currently sending these desired caps:

```
{"launchTimeout":"40000",
"platformVersion":"8.4.1",
"deviceName":"iPad4",
"processArguments":"-e AlwaysShowTutorial 1",
"noReset":true,
"appPackage":null,
"newCommandTimeout":"60",
"autoLaunch":true,
"app":null,
"platformName":"iOS"}
```

What if you changed them to:

```
{"launchTimeout":"40000",
"platformVersion":"8.4.1",
"deviceName":"iPad4",
"processArguments":["-e", "AlwaysShowTutorial", "1"],
"noReset":true,
"appPackage":null,
"newCommandTimeout":"60",
"autoLaunch":true,
"app":null,
"platformName":"iOS"}
```

**Comment by camillake** [comment]
@jlipps 
That does not work, either.
According to source code, only quotes mark added for input arguments (both string or list).
As I mentioned the format to Instrument should be  **-e flag value**.  Thus both string or list should be handled.

**Comment by MelissaHoweEmis** [comment]
I am also facing a problem regarding processArguments. The appium log doesn't show any errors when passing it in via a test and passes it in correctly but it appears to be ignored. Running the app directly from XCode with the same argument works fine. Has anyone worked out what makes the processArguments to be ignored by Appium?

**Comment by camillake** [comment]
@MelissaHoweEmis  
There s bug with processArguments. The parameter is sent to Apple instruments, but the format is not compatible with apple instruments. Thus, the processArguments sent is not accepted by Apple instruments.
It's been fixed on Appium 1.5.3 . If you don't want to upgrade. You have to modify the source code.
